solutions/communication_kafka/README.md

```diff
@@ -0,0 +1,423 @@
+   
+   
+3. Temporary/Permanent Data Storing
+   Kafka is a streaming data system that may hold data for a period of time before eliminating it. Meaning, Kafka does not discard messages as soon as the consumer reads them. 
+   
+   Kafka may also store data on permanent storage and replicate data among brokers in a cluster, giving it the aspect of a database. Kafka's data can be stored indefinitely by the user.
+
+4. Logging
+    Apache Kafka Logs are a collection of numerous data segments stored on disk that are referred to as form-topic partitions or specific topic-partitions. Each Kafka log represents a logical unique topic-based partitioning. 
+    
+    Logging with Apache Kafka makes the transformation of unstructured logging streams into an analyzable and understandable output possible. The output can then be used for detecting potential and ongoing problems as well as better environment monitoring.
+
+5. Retry 
+   Failures in a distributed system may happen, i.e. failed message process, network errors, runtime exceptions. Therefore, the retry logic implementation is something essential to have. 
+
+    It is important to note that Retries in Kafka can be quickly implemented at the consumer side. This is known as Simple Blocking Retries. To accomplish visible error handling without causing real-time disruption, Non-Bloking Retries and Dead Letter Topics are implemented.
+    
+6. Data Monitoring (Metrics)
+    Metrics are measurements that capture a value about the systems at a specific point in time, such as (provide a use case as an example from sample application). They are often collected every second, minute, or other regular interval to track a system's performance over time. 
+    
+    Kafka Monitoring Metrics measure how well a component, for example network latency, performs its role. It's vital to keep an eye on the health of Kafka deployment to ensure that the applications that rely on it keep working.
+    
+7. Event Sourcing
+   Event sourcing is a style of application design where state changes are logged as a time-ordered sequence of records. Kafka's support for very large stored log data makes it an excellent backend for an application built in this style.
+   
+   
+8. Tracking (Activity Tracking)
+   With activity tracking it is possible to track page views, searches, uploads or other actions users may take. When an event happens in the blog, for example a user logs in, an information about the event and a tracking event and will be placed into a record. The record will be placed on a specified Kafka topic.
+
+Kafka-clients have many implementations in different languages. However, this document focuses on Spring Kafka using Java.
+
+The following are the advantages and (or) disadvantages of Kafka.
```

Comment:
Even though I like this explicit list of details around kafka, this should be shortened to a bare minimum about kafka + some useful references to detailed information in blogs, documentation or whatever web resources.

I would basically describe Kafka as an event streaming plattform that stores event messages in distributed logs that form logical topics. It can be extended with a streaming API and various connectors to data sources. The main difference to other message broker is the fact that event messages are not deleted 

Maybe a simple picutre would help the unerstanding of such a small aspect and a link to the official documentation
![image](https://user-images.githubusercontent.com/40074245/173515181-3257684e-f576-4945-bdcb-8085d37a9d0d.png)
(from https://kafka.apache.org/documentation/#introduction)

Because of the focus of this article I would keep 5. Retry.